src/Ball.java

```diff
@@ -83,14 +84,52 @@ public void draw() {
      * Moves the ball so that the next time it draws it will be in a different place
      */
     public void move() {
-        x = x + xSpeed;
-        y = y + ySpeed;
-        if (x > s.width - radius || x < radius) {
-            xSpeed = -xSpeed;
-        }
-        if (y > s.height - radius || y < radius) {
-            ySpeed = -ySpeed;
+
+        /* For this code I uesed logic and hel from mr Griswold */
+        double distanceToMouse = Sketch.dist(x, y, s.mouseX, s.mouseY);
+        /*
+         * this if statements says that when the distance to mouse is greater then the
+         * radius the code for the balls to move doen't run so they stop
+         */
+        if (distanceToMouse > radius) {
+            x = x + xSpeed;
+            y = y + ySpeed;
+            if (x > s.width - radius || x < radius) {
+                xSpeed = -xSpeed;
+            }
+            if (y > s.height - radius || y < radius) {
+                ySpeed = -ySpeed;
+            }
         }
+
     }
 
-}
\ No newline at end of file
+    public boolean colliding(Ball secondBall) {
+        /*this code sets a value to the distance and then checks if the ball is colliding or not */
+        /*mr.Griswold teached me how to use the sketch.dist code in the code above */
```

Comment:
Joao I love the clarity in your explanations of learning, it's really really helpful.